src/rust/lib/graph-editor/src/component/node/port/output.rs

```diff
@@ -0,0 +1,365 @@
+
+            // === Size Change Handling == ///
+
+            eval frp.set_size ((size) data.set_size(size.into()));
+
+
+            // === Hover Event Handling == ///
+
+            port_mouse_over      <- source::<PortId>();
+            port_mouse_out       <- source::<PortId>();
+
+            delay_show_finished    <- delay_show.value.map(|t| *t>=TWEEN_END_VALUE );
+            delay_hide_finished    <- delay_hide.value.map(|t| *t>=TWEEN_END_VALUE );
+            on_delay_show_finished <- delay_show_finished.gate(&delay_show_finished).constant(());
+            on_delay_hide_finished <- delay_hide_finished.gate(&delay_hide_finished).constant(());
+
+            visible                <- delay_show_finished.map(|v| *v);
+
+            mouse_over_while_inactive  <- port_mouse_over.gate_not(&visible).constant(());
+            mouse_over_while_active    <- port_mouse_over.gate(&visible).constant(());
+
+            eval mouse_over_while_inactive ([delay_show,delay_hide](_){
+                delay_hide.stop();
+                delay_show.rewind();
+                delay_show.set_end_value(TWEEN_END_VALUE);
+            });
+            eval port_mouse_out ([delay_hide,delay_show](_){
+                delay_show.stop();
+                delay_hide.rewind();
+                delay_hide.set_end_value(TWEEN_END_VALUE);
+            });
+
+            activate_ports <- any(mouse_over_while_active,on_delay_show_finished);
+            eval_ activate_ports (delay_hide.rewind());
+
+            activate_ports_with_selected <- port_mouse_over.sample(&activate_ports);
+
+            hide_all <- on_delay_hide_finished.map(f_!(delay_show.rewind()));
+
+        }
+
+        // Init ports
+        for (index,view) in data.ports.borrow().iter().enumerate() {
+            let shape        = &view.shape;
+            let port_size    = Animation::<f32>::new(&network);
+            let port_opacity = Animation::<f32>::new(&network);
+
+            frp::extend! { network
+
+
+                // === Mouse Event Handling == ///
+
+                eval_ view.events.mouse_over(port_mouse_over.emit(PortId{index}));
+                eval_ view.events.mouse_out(port_mouse_out.emit(PortId{index}));
+                eval_ view.events.mouse_down(frp.on_port_mouse_down.emit(PortId{index}));
+
+
+                 // === Animation Handling == ///
+
+                 eval port_size.value    ((size) shape.grow.set(*size));
+                 eval port_opacity.value ((size) shape.opacity.set(*size));
+
+
+                // === Visibility and Highlight Handling == ///
+
+                 def _hide_all = hide_all.map(f_!([port_size,port_opacity]{
+                     port_size.set_target_value(0.0);
+                     port_opacity.set_target_value(0.0);
+                 }));
+
+                // Through the provided ID we can infer whether this port should be highlighted.
+                is_selected      <- activate_ports_with_selected.map(move |id| id.index == index);
```

Comment:
why this `move` is needed?

Reply:
To move `index` (laversion: `port_id`) into the closure.